.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install libssl-dev
+
+      - name: Check default-features
+        run: cargo check
+
+      - name: Check tls
+        run: cargo check --features tls
+
+      - name: Test
+        run: cargo test
+
+  autobahn:
+    name: Autobahn tests
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
```

Comment:
I'm not familiar with GitHub Actions, but maybe you can advise: is there a possibility to define rust matrix once, so that we don't have repeat the common code for different "jobs" (or "targets").

Reply:
The matrix can only defined at job level and not for the whole workflow.
See [workflow syntax][workflow-syntax] for more information.

[workflow-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy